alpha/declcfg/diff_include.go

```diff
@@ -0,0 +1,78 @@
+package declcfg
+
+import (
+	"github.com/operator-framework/operator-registry/alpha/model"
+)
+
+// makeUpgradeGraph creates a DAG of bundles with map key Bundle.Replaces.
+func makeUpgradeGraph(ch *model.Channel) map[string][]*model.Bundle {
+	graph := map[string][]*model.Bundle{}
+	for _, b := range ch.Bundles {
+		b := b
+		if b.Replaces != "" {
+			graph[b.Replaces] = append(graph[b.Replaces], b)
+		}
+	}
+	return graph
+}
+
+// findIntersectingBundles finds the intersecting bundle of start and end in the
+// replaces upgrade graph graph by traversing down to the lowest graph node,
+// then returns every bundle higher than the intersection. It is possible
+// to find no intersection; this should only happen when start and end
+// are not part of the same upgrade graph.
+// Output bundle order is not guaranteed.
```

Comment:
Non-blocking nit: it might be nice to have a simple ASCII diagram in this comment to help explain what this function is doing.

Reply:
Resolving this comment. I think the tests do a good enough job explaining it.